### PR TITLE
refactor: Update Configs

### PR DESCRIPTION
**Pull request by jshlbrd**
## Description

- Moves the Substation config so that it is public (can be imported by other Go apps)
  - `config.go` and Jsonnet functions are now in `config/`
  - example configurations are now in `examples/`
- Adds runtime concurrency setting via `SUBSTATION_CONCURRENCY` environment variable 
  - Default behavior is the same as the current behavior
- Updated documentation

## Motivation and Context
- Making the Substation config public was a requirement to make the condition inspectors and processors highly usable in other applications
- Using `SUBSTATION_CONCURRENCY` is preferred over pinning concurrency to the number of CPUs -- there are some scenarios when less or more concurrency is required

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -141,19 +141,19 @@ $ ./substation -h
 
 ### Step 2: Compile the quickstart Configuration File
 
-From the project root, run the commands below to compile the [quickstart Jsonnet configuration files](config/quickstart) into a Substation JSON config.
+From the project root, run the commands below to compile the [quickstart Jsonnet configuration files](configs/quickstart) into a Substation JSON config.
 
 ```bash
 $ sh build/config/compile.sh
```
Comment (julieagnessparks) [inline README.md:147]: Should this be `configs`?

Comment (julieagnessparks) [reply README.md:147]: Actually nvm this is correct!

**Review by julieagnessparks — APPROVED** [review]
Looks great! My only feedback is that /config vs /configs can be confusing change for others to understand which is for example and which is the actual config file.

**Comment by jshlbrd** [comment]
> Looks great! My only feedback is that /config vs /configs can be confusing change for others to understand which is for example and which is the actual config file.

I think this is a bit better now, but let me know what you think. Changes:
- All examples are now in `examples/`
- Build steps for both the quickstart (in the main README) and the AWS test deployment are simpler (copy/paste and it just works assuming the environment variables are correct)
- Added examples for using conditions and processors in external apps (examples/condition and examples/process)

**Comment by julieagnessparks** [comment]
Great changes, much easier to navigate :) LGTM 🚢

**Pull request merged**